spanner/ot_metrics.go

```diff
@@ -0,0 +1,258 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package spanner
+
+import (
+	"context"
+	"log"
+	"strconv"
+	"strings"
+	"sync"
+
+	"cloud.google.com/go/spanner/internal"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+	"google.golang.org/grpc/metadata"
+)
+
+// OtInstrumentationScope is the instrumentation name that will be associated with the emitted telemetry.
+const OtInstrumentationScope = "cloud.google.com/go"
+const metricsPrefix = "spanner/"
+
+var (
+	attributeKeyClientID   = attribute.Key("client_id")
+	attributeKeyDatabase   = attribute.Key("database")
+	attributeKeyInstance   = attribute.Key("instance_id")
+	attributeKeyLibVersion = attribute.Key("library_version")
+	attributeKeyType       = attribute.Key("type")
+	attributeKeyMethod     = attribute.Key("grpc_client_method")
+
+	attributeNumInUseSessions = attributeKeyType.String("num_in_use_sessions")
+	attributeNumSessions      = attributeKeyType.String("num_sessions")
+	// openTelemetryMetricsEnabled is used to track if OpenTelemetry Metrics need to be recorded
+	openTelemetryMetricsEnabled = false
+	// mutex to avoid data race in reading/writing the above flag
+	otMu = sync.RWMutex{}
+)
+
+func getOpenTelemetryConfig(mp metric.MeterProvider, logger *log.Logger, sessionClientID string, db string) (*openTelemetryConfig, error) {
+	logf(logger, "Inside getOpenTelemetryConfig()")
```

Comment:
This seems like a left-over from debugging?

Reply:
Removed